x/nft/action.go

```diff
@@ -0,0 +1,75 @@
+package nft
+
+import (
+	"fmt"
+	"regexp"
+	"sync"
+)
+
+// Action represents available and supported by the implementation actions.
+// This is just a string type alias, but using it increase the clarity of the
+// API.
+type Action string
+
+// nft package provides default set of actions.
+const (
+	UpdateDetails   Action = "ActionUpdateDetails"
+	Transfer        Action = "ActionTransfer"
+	UpdateApprovals Action = "ActionUpdateApprovals"
+)
+
+func init() {
+	RegisterAction(UpdateDetails)
```

Comment:
Since we have a vararg here why don't we register all of them in one line?
I would also say let's not use init, but instead resort to something more explicit on the app level.

I see where you are coming from as you want to have default actions registered, but I'd rather not use init for this, because this reduces the flexibility and testability of the system, also predictability. 

We are not using init() blocks anywhere in our code for this reason.

What I would propose is registering the actions within the app, that would also provide a good base to go forward for other developers when they need to register their custom actions.

What I would do is create a list in nft package defaultActions and do RegisterAction(defaultActions...) within bns app.go. 

@alpe @ethanfrey maybe you want to chip in on this for second opinion?

Reply:
I think with this approach it's only different trades that we can do, none of which leads to great code.

Default actions are required by the application, so they must be registered, even if user does not register any. If they are mandatory, why do we force user to register them?

I want to use `const`, because it seems right, with `var` it could be done without `init` function:
```go
var (
   Foo = RegisterAction("Foo")
   Bar = RegisterAction("Bar")
)
```
I think using `const` is a better choice.

Testing is broken in general, because to not over complicate it, actions register is global. If you modify it in one test, this modification is carries throughout all tests. This is terrible but in this case I also don't know how could it be done better.

Reply:
@husio implement a Clear method on the register and it's already better!
Let's use a per app approach for registering things as that is what we do everywhere
else, so then at least we are consistent. Default actions are required by the module and not the application, so it is important to realise that we want to have explicit control over what is registered and when per application, so I'd prefer to have something explicit rather than init or constants which register things automatically. 

One thing I can propose is registering these in base controller since they are indeed required by the nft module and so is the controller.

I think we had this discussion long ago with @alpe  and/or @ethanfrey and were against automagic registration of things. 

I would really love to hear a second opinion on that one, because I do realise the benefits of not having to register things by hand.

Reply:
> What I would do is create a list in nft package defaultActions and do RegisterAction(defaultActions...) within bns app.go.

I agree this sounds like a better approach than `init` function as it gives us most flexibility and separates framework and app.